mdata/store_cassandra.go

```diff
@@ -204,9 +204,16 @@ func (c *cassandraStore) processWriteQueue(queue chan *ChunkWriteRequest, meter
 
 			data := cwr.chunk.Series.Bytes()
 			chunkSizeAtSave.Value(int64(len(data)))
-			version := chunk.FormatStandardGoTsz
+			version := chunk.FormatStandardGoTszWithSpan
 			buf := new(bytes.Buffer)
-			binary.Write(buf, binary.LittleEndian, uint8(version))
+			binary.Write(buf, binary.LittleEndian, chunk.Format(version))
+
+			chunkSize, ok := chunk.RevChunkSpans[cwr.span]
+			if !ok {
+				// it's probably better to panic than to persist the chunk with a wrong length
+				panic(fmt.Sprintf("Chunk size invalid: %d", cwr.span))
```

Comment:
actually it's one of those things that `should never happen` because at the time this `span` gets configured there is already a check like that: https://github.com/raintank/metrictank/pull/418/files#diff-e33a682d2077aab8454c0e304dc7fe3dR253